docs/introduction-to-smart-contracts.rst

```diff
@@ -207,8 +206,7 @@ The functions that make up the contract, and that users and contracts can call a
 
 The ``mint`` function sends an amount of newly created coins to another address.
 The :ref:`require <assert-and-require>` function call defines conditions that reverts all changes if not met.
-In this example, ``require(msg.sender == minter);`` ensures that only the creator of the contract can call ``mint``,
-and ``require(amount < 1e60);`` ensures a maximum amount of tokens. This ensures that there are no overflow errors in the future.
+In this example, ``require(msg.sender == minter);`` ensures that only the creator of the contract can call ``mint``. Note that because of the default :ref:`Checked arithmetic <unchecked>`, the transaction would revert if the expression ``balances[receiver] += amount;`` overflows, i.e., when ``balances[receiver] + amount`` in arbitrary precision arithmetic is larger than the maximum value of ``uint`` (``2**256 - 1``). This is also true for the statement ``balances[receiver] += amount;`` in the function ``send``.
```

Comment:
Please add a paragraph in the rendered document and add line breaks in the source code.

Maybe we should also introduce this sentence with something like `In general, the creator can mint as many tokens as they like, but at some point, this will lead to a phenomenon called "overflow":`

Reply:
Note that this is the "introduction to smart contracts" sections - we assume readers have not ever seen a smart contract before in their life, or not even a program in general.